### ISSUE TITLE
Fixes for VPAT/admin area

### ISSUE THREAD

**Issue by sfisher** [CDL-Dryad/dryad-product-roadmap#1373]
See https://docs.google.com/spreadsheets/d/14PwHqY7rnlEAemad-Dsn0VNADFODudTZ9q-oVfu6Ffc/edit#gid=545873482 

- [x] ACTIVITY LOG: Add Data modal’s form fields are not associated correctly with their labels.
- [x] Curation list: The edit status modal (dialog window) does not have an informative title.
- [x] The edit status modal’s “Notes” and “Status” fields are not associated correctly with their labels.
- [x] Pagination links are not labeled. They are read as numbers.
- [x] VoiceOver skips columns containing action icon buttons when reading row 1 in table.
